### ISSUE TITLE
Map db.Text to TEXT in SQLite

### ISSUE THREAD

**Issue by cannikin** [prisma/prisma#9012]
## Problem

Even though SQLite's native data type for strings is `TEXT`, trying to actually use `@db.Text` as the native data type results in an error: "Native type Text is not supported for sqlite connector." This is counter-intuitive since the two are literally named the same thing and SQLite only has one string-like datatype, `TEXT`.

## Suggested solution

Map both `String` and native type `db.Text` to `TEXT` in SQLite.

## Additional context

It seems reasonable to allow a developer to be more expressive in the schema, declaring their intent for these fields. Traditionally, in the database world, a "string" would default to `VARCHAR` (which has a max of 255 characters) so if you want to allow more, you need to use `TEXT`. This isn't necessary in SQLite, but it can still help to visually express that you expect a certain field to contain lots of text.

This is also helpful if/when someone decides to move from SQLite to another database: no need to go through and add more native types since they're already in place.

**Comment by janpio** [comment]
Prisma has a set of scalar types that we strive to support for all databases (where possible).
Additionally we allow our users to customize the types of created fields with native database types.
(Examples: https://www.prisma.io/docs/concepts/database-connectors/postgresql#type-mapping-between-postgresql-to-prisma-schema or https://www.prisma.io/docs/concepts/database-connectors/mysql#type-mapping-between-mysql-to-prisma-schema)

For SQLite we do not support _any_ native database types, as all the available types are covered by scalar types: https://www.prisma.io/docs/concepts/database-connectors/sqlite#data-model-mapping

We could now allow all the native types that are already mapped by the scalar types - would that satisfy the expressiveness?
Alternatively we could make the error message a bit less intimidating and explain why SQLite does not have native types instead.

**Comment by cannikin** [comment]
> We could now allow all the native types that are already mapped by the scalar types

I think so? If I understand this correctly: since SQLite is already using `TEXT` natively, you should be able to write:

```
model Comment {
  id      @id    @default(autoincrement())
  message String @db.Text
}
```

And not get an error (since the type is *already* TEXT in SQLite). Is that what you were saying? If so yes, that would be awesome. :)

**Comment by janpio** [comment]
Yes, that is indeed what I meant and what we could do. We are not fully certain if it is a good idea yet.
